buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt

```diff
@@ -244,6 +245,44 @@ class PerformanceTestPlugin : Plugin<Project> {
             channel = "adhoc"
             outputs.doNotCacheIf("Is adhoc performance test") { true }
         }
+
+        val shouldLoadScenariosFromFile = project.providers.gradleProperty("includePerformanceTestScenarios")
+            .forUseAtConfigurationTime()
+            .getOrElse("false") == "true"
+
+        configureSampleGenerators {
+            all {
+                val sampleGenerator = this
+                create("${sampleGenerator.name}PerformanceAdHocTest") {
+                    performanceReporter = createPerformanceReporter()
+                    channel = "adhoc"
+                    outputs.doNotCacheIf("Is adhoc performance test") { true }
+                    setTestProjectGenerationTask(sampleGenerator)
+                }
+                val channelSuffix = if (OperatingSystem.current().isLinux) "" else "-${OperatingSystem.current().familyName.toLowerCase()}"
+                create("${sampleGenerator.name}PerformanceTest") {
+                    performanceReporter = createPerformanceReporter()
+                    channel = "commits$channelSuffix"
+                    setTestProjectGenerationTask(sampleGenerator)
+
+                    if (shouldLoadScenariosFromFile) {
+                        scenarios = loadScenariosFromFile(sampleGenerator.name).joinToString(";")
+                        doFirst {
+                            assert(scenarios.isNotEmpty()) { "Running $name requires to set some scenarios" }
```

Comment:
Does `$name` here refer to `sampleGenerator.name`?

Reply:
No, it refers to the task name.

Reply:
What does `sampleGenerator.name` refer to?

Reply:
Let's do an example:
* `sampleGenerator.name == largeJavaMultiProject`
* `name == largeJavaMultiProjectPerformanceTest`.